### PR TITLE
fix: add Auth context cache

### PR DESCRIPTION
**Pull request by chasers**
- [x] adds Auth.Cache context cache
- [x] Calls Auth.Cache in VerifyApiAccess plug
- [x] Expires Auth.Cache set to 30 seconds (this cache doesn't get busted from the write ahead log)
- [x] Current tests pass
- [x] Tests for Auth.Cache

### CODE REVIEW

**Review by hauleth — COMMENTED** [review]
Do we have any possibility to revoke existing token? Like for example removing user from organisation or forceful session end due to security breach. If so, how does it work with that cache?

**Comment by chasers** [comment]
> Do we have any possibility to revoke existing token? Like for example removing user from organisation or forceful session end due to security breach. If so, how does it work with that cache?

Just have to give it 30 seconds and the cache key for that user will expire, failing down to calling the fn again.

File: test/logflare/auth/cache_test.exs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+defmodule Logflare.Auth.CacheTest do
+  use Logflare.DataCase
+  alias Logflare.Auth
+  alias Logflare.Factory
```
Comment (Ziinc) [inline test/logflare/auth/cache_test.exs:4]: should be auto imported already

File: test/logflare/auth/cache_test.exs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  alias Logflare.Partners.Partner
+
+  setup do
+    user = Factory.insert(:user)
+    [user: user, team: Factory.insert(:team, user: user), partner: Factory.insert(:partner)]
```
Comment (Ziinc) [inline test/logflare/auth/cache_test.exs:9]:
```suggestion
    [user: user, team: insert(:team, user: user), partner: insert(:partner)]
```

File: test/logflare/auth/cache_test.exs [hunk 3]
```diff
@@ -0,0 +1,55 @@
+defmodule Logflare.Auth.CacheTest do
+  use Logflare.DataCase
+  alias Logflare.Auth
+  alias Logflare.Factory
+  alias Logflare.Partners.Partner
+
+  setup do
+    user = Factory.insert(:user)
```
Comment (Ziinc) [inline test/logflare/auth/cache_test.exs:8]:
```suggestion
    user = insert(:user)
```

**Review by Ziinc — APPROVED** [review]

**Comment by Ziinc** [comment]
merging in to get this to prod.

**Pull request merged**